SPEC.md

```diff
@@ -0,0 +1,675 @@
+In cases of attributable faults, a watcher can submit a slash proof on-chain as evidence of a transcoder's malicious behavior. In cases of non-attributable faults, a transcoder's behavior can not be definitively considered
+malicious due to the possible failure of external dependencies.
+
+Attributable faults include:
+
+- **Missed verification**: The transcoder did not submit transcode proofs for challenged segments.
+- **Double claiming segments**: The transcoder claimed a segment more than once.
+
+Non-attributable faults include:
+
+- **Failed verification**: The transcoder failed verification for a challenged segment. This type of fault is non-attributable because the protocol cannot differentiate from the scenario in which
+the transcoder failed verification because it submitted an incorrect transcoded data hash with its transcode proof and the scenario in which the transcoder
+honestly submitted a correct transcoded data hash, but the external verifier submitted an incorrect result or failed to submit a result at all.
+
+In the initial version of the protocol, all faults are penalized equally. However, in the future, non-attributable faults could be penalized less serverely than attributable faults.
+
+## Contract API
+
+See the [developer documentation](#) for the full documentation of the contract API and descriptions of the contract functions described in this specification.
+
+## Actions
+
+This section describes the state transitions associated with common protocol actions.
+
+### Round Progression
+
+#### Requirements
+
+At the start of each round, the round must be initialized before any other protocol actions can be executed. Round initialization entails
+locking in the latest active transcoder set for the round and setting the maximum number of tokens that can be issued (based on the current inflation rate) and redistributed (from a redistribution pool) during the round.
+A round can only be initialized once and anybody can invoke the function to initialize a round.
+
+#### Initial State
+
+- `B` is the current block number such that `B % roundLength == 0`
+
+#### State Affected
+
+- `RoundsManager`
+- `BondingManager`
+- `Minter`
+
+#### Algorithm
+
+1. At some block number `M` in the set `B, B+1, B+2...B + roundLength - 1`, a user calls `roundsManager.initializeRound()`.
+2. `currentRound = M / roundLength`.
+3. If the last initialized round is not less than `currentRound`, abort.
+3. Set the last initialized round to `currentRound`.
+4. Take the first `numActiveTranscoders` transcoders from the transcoder pool and set them as the members of the active transcoder set for the round.
+5. Set the maximum number of LPT that can be issued in the round based on the current LPT supply and the current `inflationRate`.
+6. Set the maximum number of LPT that can be redistributed for the round based on the current redistribution pool and some constant number of rounds to spread the redistribution across.
+
+#### Notes
```

Comment:
One idea for the client here is that perhaps the client could have logic to randomly call initialize round with a certain probability if you believe you're in the transcoder set. And then it could sleep for awhile and re-attempt if it wasn't yet initialized by someone. Just an idea in general, not necessarily for the spec.